applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -284,53 +251,81 @@ def _SetNodalProperties(self):
+            self.GetComputingModelPart(),
+            maximum_multiple_of_g_acceleration_allowed)
+
+        return acceleration_limitation_utility
+
+    def _CreateLevelSetConvectionProcess(self):
         # Construct the level set convection process
+        domain_size = self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+        computing_model_part = self.GetComputingModelPart()
         if self._bfecc_convection:
             if have_conv_diff:
-                if self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
-                    locator = KratosMultiphysics.BinBasedFastPointLocator2D(self.main_model_part).UpdateSearchDatabase()
+                if domain_size == 2:
+                    locator = KratosMultiphysics.BinBasedFastPointLocator2D(computing_model_part).UpdateSearchDatabase()
                     level_set_convection_process = KratosConvDiff.BFECCConvection2D(locator)
                 else:
-                    locator = KratosMultiphysics.BinBasedFastPointLocator3D(self.main_model_part).UpdateSearchDatabase()
+                    locator = KratosMultiphysics.BinBasedFastPointLocator3D(computing_model_part).UpdateSearchDatabase()
                     level_set_convection_process = KratosConvDiff.BFECCConvection3D(locator)
             else:
                 raise Exception("The BFECC level set convection requires the Kratos ConvectionDiffusionApplication compilation.")
         else:
-            if self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
+            linear_solver = self._GetLinearSolver()
+            if domain_size == 2:
                 level_set_convection_process = KratosMultiphysics.LevelSetConvectionProcess2D(
                     KratosMultiphysics.DISTANCE,
-                    self.main_model_part,
-                    self.linear_solver)
+                    computing_model_part,
+                    linear_solver)
             else:
                 level_set_convection_process = KratosMultiphysics.LevelSetConvectionProcess3D(
                     KratosMultiphysics.DISTANCE,
-                    self.main_model_part,
-                    self.linear_solver)
+                    computing_model_part,
+                    linear_solver)
 
         return level_set_convection_process
 
-    def _set_variational_distance_process(self):
+    def _CreateVariationalDistanceProcess(self):
         # Construct the variational distance calculation process
         maximum_iterations = 2 #TODO: Make this user-definable
+        linear_solver = self._GetLinearSolver()
+        computing_model_part = self.GetComputingModelPart()
         if self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
             variational_distance_process = KratosMultiphysics.VariationalDistanceCalculationProcess2D(
-                self.main_model_part,
-                self.linear_solver,
+                computing_model_part,
```

Comment:
hm you sure abt this?
Maybe for a different PR?

Reply:
I'm sure. The level set needs only to be computed in the computing model part, which is the one used in the CFD problem resolution.

Reply:
I ran the application tests and the ones in the Examples repo (in serial and MPI).